server/src/main/scala/org/broadinstitute/workbench/hamm/server/WorkflowCostService.scala

```diff
@@ -1,50 +1,73 @@
 package org.broadinstitute.workbench.hamm
 package server
 
-import java.time.Instant
+import java.util.UUID
 
 import cats.effect.Sync
 import cats.implicits._
 import io.chrisdavenport.log4cats.Logger
 import io.grpc.Metadata
-import org.broadinstitute.workbench.hamm.dao.GooglePriceListDAO
+import org.broadinstitute.workbench.hamm.auth.HttpSamDAO
+import org.broadinstitute.workbench.hamm.dao.{GooglePriceListDAO, WorkflowMetadataDAO}
 import org.broadinstitute.workbench.hamm.model._
 import org.broadinstitute.workbench.hamm.protos.hamm._
 
-class WorkflowCostService[F[_]: Sync: Logger](pricing: GooglePriceListDAO[F]) extends HammFs2Grpc[F] {
+class WorkflowCostService[F[_]: Sync: Logger](pricing: GooglePriceListDAO[F],
+                                              workflowDAO: WorkflowMetadataDAO[F],
+                                              samDAO: HttpSamDAO[F]) extends HammFs2Grpc[F] {
+
   override def getWorkflowCost(request: WorkflowCostRequest, clientHeaders: Metadata): F[WorkflowCostResponse] = {
+    withAuthenticatedUser(clientHeaders) { userInfo =>
+      //ToDo: some work here to make this less messy
+      for {
+        cromwellMetadata: MetadataResponse <- workflowDAO.getMetadata(WorkflowId(UUID.fromString(request.id)))
```

Comment:
if `: MetadataResponse` is not needed, it's better to remove it...cuz it'll add an additional function call when things are desugared